yb_migrate/src/srcdb/source.go

```diff
@@ -104,3 +130,108 @@ func (s *Source) ParseURI() {
 
 	}
 }
+
+func parseSSLString(source *Source) {
+	if source.SSLQueryString == "" {
+		return
+	}
+
+	validParams := []string{"sslmode", "sslcert", "sslrootcert", "sslkey"}
+
+	sslParams := strings.Split(source.SSLQueryString, "&")
+	for _, param := range sslParams {
+		slicedparam := strings.Split(param, "=")
+		for i, checkParam := range validParams {
+			if checkParam == slicedparam[0] {
+				switch i {
+				case 0:
+					source.SSLMode = slicedparam[1]
+				case 1:
+					source.SSLCertPath = slicedparam[1]
+				case 2:
+					source.SSLRootCert = slicedparam[1]
+				case 3:
+					source.SSLKey = slicedparam[1]
+				}
+				break
+			}
+		}
+	}
+}
+
+func createTLSConf(source *Source) tls.Config {
+	rootCertPool := x509.NewCertPool()
+	if source.SSLRootCert != "" {
+		pem, err := ioutil.ReadFile(source.SSLRootCert)
+		if err != nil {
+			ErrExit("error in reading SSL Root Certificate: %v", err)
+		}
+
+		if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+			ErrExit("Failed to append PEM.")
+		}
+	} else {
+		ErrExit("Root Certificate Needed for verify-ca and verify-full SSL Modes")
+	}
+	clientCert := make([]tls.Certificate, 0, 1)
+
+	if source.SSLCertPath != "" && source.SSLKey != "" {
+		certs, err := tls.LoadX509KeyPair(source.SSLCertPath, source.SSLKey)
+		if err != nil {
+			ErrExit("error in reading and parsing SSL KeyPair: %v", err)
+		}
+
+		clientCert = append(clientCert, certs)
+	}
+
+	if source.SSLMode == "verify-ca" {
+		return tls.Config{
+			RootCAs:            rootCertPool,
+			Certificates:       clientCert,
+			InsecureSkipVerify: true,
+		}
+	} else { //if verify-full
+
+		return tls.Config{
+			RootCAs:            rootCertPool,
+			Certificates:       clientCert,
+			InsecureSkipVerify: false,
+			ServerName:         source.Host,
+		}
+	}
+}
+
+func generateSSLQueryStringIfNotExists(s *Source) string {
```

Comment:
nit: it is postgres specific, can be moved to `srcdb/postgres.go`

Reply:
Same as above.